---
layout: page
title: Joe Bielawski
permalink: /faculty-bielawski/
---
{% include figure.html description="Joe" url="/faculty/bielawski/img/JPB.jpg" css="image-left" height="250px" %}
**Arrival:** Not yet determined.

**Departure:** Not yet determined.

**Institutional home pages:**

[Statistics](http://www.mscs.dal.ca/Faculty/bielawski.html)

[Biology](http://www.bielawski.info)

## Lecture Materials
{:style="clear:both"}

I did an extensive update of the lecture slides on codon models for 2017. The update includes greater coverage of the mechanistic process of codon evolution (via the MutSel framework).  Because there is now less detail about fitting codon models to real data, I have included links to the 2015 & 2016 slides below; these slides provide more practical information about the powers and pitfalls of inference under codon models.

{% comment %}
Note that going to the **schedule page** and using the links there will result in one slide per page, which is **easier to follow during the lecture**.
{% endcomment %}

The **links below** will download files with two slides per page, and these are **better for printing and note-taking**.

**2019 Lecture slides, Parts 1 & 2**: [PDF file1 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2019_Parts1and2_handouts.pdf)

**2019 Lecture slides, Parts 3 & 4**: [PDF file2 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2019_Parts3and4_handouts.pdf)


**2018 Lecture slides, Parts 1 & 2**: [PDF file1 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2018_Parts1and2_Handouts.pdf)

**2018 Lecture slides, Parts 3 & 4**: [PDF file2 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2018_Part3and4_handouts.pdf)


**2017 Lecture slides, Parts 1 & 2**: [PDF file1 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2017_Parts1and2.pdf)

**2017 Lecture slides, Part 3**: [PDF file2 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2017_Part3.pdf)

**2017 Lecture slides, Part 4**: [PDF file3 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2017_Part4.pdf)


**2016 Lecture slides, Part 1**: [PDF file1 (1 slide per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2016_lecture_Parts1and2_FullPage.pdf)

**2016 Lecture slides, Part 2**: [PDF file2 (1 slide per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/WH2016_lecture_Part3.pdf)

**2015 Lecture slides, Part 1**: [PDF file1 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/Lecture_WH2015_Part1_2slidesPage.pdf)

**2015 Lecture slides, Part 2**: [PDF file2 (2 slides per page)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/slides/Lecture_WH2015_Part2_2slidesPage.pdf)


## Key papers related to the Lectures

Phenomenological load (PL) and biological conclusions under codon models:<br>
(_[Jones C.T., Youssef N., Susko E., Bielawski J.P., 2018. Phenomenological Load on Model Parameters Can Lead to False Biological Conclusions. Mol Biol Evol. 35(6):1473-1488.](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/Jones_PL_MBE2018.pdf)_)


Review of major inference challenges under codon models:<br>
(_[Jones C.T., Susko E., Bielawski J.P.,  2018. Looking for Darwin in genomic sequences: validity and success depends on the relationship between model and data. In Evolutionary Genomics: Statistical and Computational Methods. Maria Anisimova (ed.) 2nd edition (In Press), Human press.](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/JonesSuskoBielawski_InPress.pdf)_)

Positive selection, purifying selection, shifting balance & fitness landscapes:<br>
(_[Jones, C., Youssef, N., Susko, E. and Bielawski, J., 2017.  Shifting balance on a static mutation-selection landscape: a novel scenario of positive selection. Molecular Biology and Evolution, 34(2):391-407.](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/ShiftBalance_MBE_2016.pdf)_)

Improved inference of site-specific positive selection under a generalized parametric codon model when there are
multinucleotide mutations and multiple nonsynonymous rates: <br>
(_[Dunn KA, Kenney T, Gu H, Bielawski JP. Improved inference of site-specific positive selection under a generalized parametric codon model when there are multinucleotide mutations and multiple nonsynonymous rates. BMC Evol Biol. 2019 Jan 14;19(1):22.](https://bmcevolbiol.biomedcentral.com/articles/10.1186/s12862-018-1326-7)_)

ModL: restoring regularity when testing for positive selection: <br>
(_[Mingrone J, Susko E, Bielawski JP. ModL: exploring and restoring regularity when testing for positive selection. Bioinformatics. 2019 Aug 1;35(15):2545-2554.](https://academic.oup.com/bioinformatics/article/35/15/2545/5239994)_)

Smoothed Bootstrap Aggregation (SBA) for assessing and correcting paramater estimate uncertainty in codon models:<br>
(_[Mingrone, J., Susko, E. and Bielawski, J., 2016. Smoothed bootstrap aggregation for assessing selection pressure at amino acid sites. Molecular Biology and Evolution, 33(11):2976-2989.](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/Mingrone_SBA_MBE2016.pdf)_)

Protocols, experimental design, and best practices for inference under complex codon models:<br>
(_[Bielawski, J.P., Baker, J.L. and Mingrone, J., 2016. Inference of episodic changes in natural selection acting on protein coding sequences via CODEML. Current Protocols in Bioinformatics, pp.6-15.](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/UNIT6.15.pdf)_)

## PAML lab Materials

The lab exercises (PAML demo) have been published to a small website (link below). The site contains some additional resources. Please note that presentations may change a little prior to the lab. I will post modified PDFs as required. 

**PAML demo**: [http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/lab.html Link to the new website]

**PAML demo resources**: [http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/Resources.html Link]

**PAML demo slides**: [PDF file (updated slides for 2018)](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/pamlDEMO_slides.pdf)

Lab materials can also be downloaded from a repository: [https://bitbucket.org/EvoWorks/protocol-paml-lab-at-mbl-workshop/downloads Bitbucket repository]


## Supplemental PAML lab notes

[https://molevol.mbl.edu/index.php/PAML_notes Additional notes for the Lab as carried out at this workshop]

[https://molevol.mbl.edu/images/e/ed/PAML_tree_tutorial.pdf A tutorial on how to identify specific branches in a tree file]

[http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/BranchMarkedTree.jpg Illustration of a tree with specific branches marked]

**PAML demo:** [Link to the old website](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab_old/lab.html)

## Alternative Lab

If you have some experience with codon models, and want to try out a tutorial for more advanced materials then use the link below to download an archive for a complete different set of PAML activities.  This tutorial focuses on detecting episodic protein evolution via Branch-Site Model A.  The tutorial also includes activities about (i) detecting MLE instabilities, (ii) carrying out robustness analyses, and (iii) use of smoothed bootstrap aggregation (SBA).  The protocols for each activity are presented in Protocols in Bioinformatics UNIT 6.15.  The included PDF file for UNIT 6.16 also presents recommendations for "best practices" when carrying out a large-scale evolutionary survey for episodic adaptive evolution by using PAML.  The files required for this "alternative lab" are available via Bitbucket repository.  The repository link is given below.

**Advanced PAML demo:** [Bitbucket repository](https://bitbucket.org/EvoWorks/protocol-inference-of-episodic-selection/downloads)

## "Best practices" in large-scale evolutionary surveys

Large-scale evolutionary surveys are now commonplace. But with the use of progressively more complex codon models, these surveys are fraught with perils. Complex models are more prone to statistical problems such as MLE irregularities, and some can be quite sensitive to model misspecification.  UNIT 6.16 (see above) provides some recommended "best practices" for a **2-phase approach** to quality control and robustness in evolutionary surveys.  We have applied these to a large scale survey for functional divergence in nuclear receptors during homing evolution, and we used experimental approaches to investigate hypotheses about the role of a particular nuclear receptor (NR2C1) as a key modulator of developmental pluripotnetiality during hominid evolution. The paper that illustrates the power of such an evolutionary surgery, and the importance of an experimental design having explicit protocols for "best practices", is given below.

**Example large-scale survey**: [PDF reprint](http://awarnach.mathstat.dal.ca/~joeb/PAML_lab/resources/Baker.905.full.pdf)


## Alternative software for codon models in the ML framework

**HyPhy:**  comparative sequence analysis using stochastic evolutionary models; [http://www.hyphy.org/](http://www.hyphy.org/)<br>
**DataMonkey:**  a server that supports a variety of HYPHY tools at no cost; [http://www.datamonkey.org/](http://www.datamonkey.org/)<br>
**COLD:**  a program that implements a general-purpose parametric (GPP) codon model. Most codon models are special cases of the GPP codon model. [https://github.com/tjk23/COLD](https://github.com/tjk23/COLD)<br>
**codeml_SBA:**  a program that implements smoothed Bootstrap Aggregation for Assessing Selection Pressure at Amino Acid Sites.[https://github.com/Jehops/codeml_sba](https://github.com/Jehops/codeml_sba)<br>
**ModL:**  a program for restoring regularity when testing for positive selection using codon models [https://github.com/jehops/codeml_modl](https://github.com/jehops/codeml_modl)

{% include figure.html description="The Kidd" url="/faculty/bielawski/img/TheKidd1b.jpg" css="image-center" height="250px" %}
.