---
layout: page
title: L. Lacey Knowles
permalink: /faculty-knowles/
---
{% include figure.html description="Lacey Knowles" url="/faculty/knowles/img/Lacey-MOLjpg.jpg" css="image-right" height="250px" %}
### Institutional home page 

[lsa.umich.edu/eeb/people/faculty/knowlesl.html](https://lsa.umich.edu/eeb/people/faculty/knowlesl.html)

### Arrival and Departure 

May 31 to June 7

{% comment %}
### Lecture Materials

Lacey did not make her presentation available due to concerns about unpublished material.
{% endcomment %}

### Demo and Tutorials

### References 

Transformative potential of model-based analyses: examples from species delimitation, biogeography, phylogeography, and beyond the species tree

#### Species delimitation

Leache et al. (2018) [The spectre of too many species.](https://doi.org/10.1093/sysbio/syy051) Syst. Biol., in press. 

Jackson ND, Morales A, Carstens BC, O'Meara BC (2017) [Species delimitation with gene flow.](http://doi.org/10.1093/sysbio/syw117) Systematic Biology 66, 799-812. 

Sukumaran J, Knowles LL (2017) Multispecies coalescent delimits structure, not species. Proc. Natl. Academy Sci. USA 114:1607-1612.

Yang Z, Rannala B (2010) [Bayesian species delimitation using multilocus sequence data.](https://doi.org/10.1073/pnas.0913022107) PNAS 107:9264-9269.

#### Biogeography

Matzke, N.J. (2012) Founder-event speciation in biogeobears package dramatically improves likelihoods and alters parameter inference in dispersal-extinction-cladogenesis (DEC) analyses. Front. Biogeogr. 4, 210

Ree, R.H. and Sanmartin, I. (2018) [Conceptual and statistical problems with the DEC+ J model of founder-event speciation and its comparison with DEC via model selection.](http://dx.doi.org/10.1111/jbi.13173) J. Biogeogr. 45, 741-749. 

Ree, R.H. and Smith, A.A. (2008) Maximum likelihood inference of geographic range evolution by dispersal, local extinction, and cladogenesis. Syst. Biol. 57, 4–14

Ronquist, F. and Sanmartn, I. (2011) Phylogenetic methods in biogeography. Annu. Rev. Ecol. Evol. Syst. 42, 441–464

Sukumaran J, Economo EP, Knowles LL (2016) Machine learning biogeographic processes from biotic patterns: a new trait-dependent dispersal and diversification model, with model-choice by simulation-trained discriminant analysis. Syst. Biol. 65, 525-545.

Sukumaran J, Knowles LL (2018) Trait-dependent biogeography: (Re-)integration of biology into probabilistic historical biogeographical models. Trends Ecol. Evol. 33, 390-398.

#### Phylogeography

Carstens et al. (2017) Objective choice of phylogeographic models. Mol. Phylog. Evol. 116: 136-140.

He Q, Prado JR, Knowles LL (2017) [Inferring the geographic origin of a range expansion: latitudinal and longitudinal coordinates inferred from genomic data in an ABC framework with the program X-ORIGIN.](http://doi.org/10.1111/mec.14380) Mol. Ecol. 26:6908-6920.

Massatti R, Knowles LL (2016) Contrasting support for alternative models of genomic variation based on microhabitat preference: species-specific effects of climate change in alpine sedges. Mol. Ecol. 25:3974-3986.

Oaks, J. R., J. Sukumaran, J. A. Esselstyn, C. W. Linkem, C. D. Siler, M., T. Holder, and R. M. Brown. 2013. Evidence for climate-driven diversification? A caution for interpreting ABC inferences of simultaneous historical events. Evolution 67:991–1010.

Papadopoulou A, Knowles LL (2016) Towards a paradigm shift in comparative phylogeography driven by trait-based hypotheses. Proc. Natl. Academy Sci. USA 113:8018-8024.

#### Beyond the species tree

Huang J-P, Knowles LL (2018) Testing the impact of oceanic barriers on population subdivision, speciation, and zoogeographic community assembly in Xylotrupes beetles across the Indo-Australian Archipelago. Biol. J. Linnean Soc., in press.

Knowles LL, Huang H, Sukumaran J, Smith SA, (2018) A matter of phylogenetic scale: distinguishing incomplete lineage sorting from lateral gene transfer as the cause of gene tree discord in recent versus deep diversification histories. Am. J. Botany 105:376-384.

Smith, S. A., M. Moore, J. W. Brown, and Y. Yang (2015). Analysis of phylogenomic datasets reveals conflict, concordance, and gene duplications with examples from animals and plants. BMC Evolutionary Biology 15:150
